### PR TITLE
Add Python 3 to the build matrix.

### PR DESCRIPTION
**Pull request by zenhack**
There is predicatbly a failure, but I suspect it's something actually related to Python 3 now, so I'm submitting this as a pr.

Relates to: https://github.com/CCI-MOC/hil/issues/700

### CODE REVIEW

**Comment by xuhang57** [comment]
I might not know what I am talking about. But fwiw, normally, python 3 is using pip3 rather than pip2 to manage packages so does that mean we need to update that in travis for our python 3 build?

**Comment by naved001** [comment]
Didn't make a difference when I was using pip3 vs pip for installing in my python3 virtualenv. it just couldn't import `importlib._bootstrap` in both cases (the current failure in our travis build).

**Comment by naved001** [comment]
Also, in a virtualenv with python3 as default, pip seems to refer to pip3.

```
(.venv) naved:~/hil/.venv/bin$ python --version
Python 3.5.2
(.venv) naved:~/hil/.venv/bin$ pip --version
pip 9.0.1 from /home/naved/hil/.venv/lib/python3.5/site-packages (python 3.5)
(.venv) naved:~/hil/.venv/bin$ pip3 --version
pip 9.0.1 from /home/naved/hil/.venv/lib/python3.5/site-packages (python 3.5)
(.venv) naved:~/hil/.venv/bin$ pip2 --version
pip 9.0.1 from /usr/local/lib/python2.7/dist-packages (python 2.7)
```

**Comment by xuhang57** [comment]
Is there a way to let travis to show all builds on this page?

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,10 @@
 language: python
 python:
   - "2.7"
+  - "3.4"
+matrix:
+  allow_failures:
+    - python: "3.4"
```
Comment (xuhang57) [inline .travis.yml]: I know you are using `3.4` because `3.6` is failing. But since `3.4` is failing as well, should we just use `3.6`?

**Comment by xuhang57** [comment]
https://devguide.python.org/#status-of-python-branches

Given what it says in the official guide, we should start with `3.6` since `3.4` will be deprecated even before `2.7`

**Comment by zenhack** [comment]
Quoting Lucas H. Xu (2018-03-05 16:38:23)

>    I know you are using 3.4 because 3.6 is failing. But since 3.4 is
>    failing as well, should we just use 3.6?

We expect failures from any 3.x variant, but the 3.6 failures look like
they're not even getting as far as *installing* python 3.6. I suspect
the current failure is an actual portability bug, which is a bit better.

**Comment by xuhang57** [comment]
@zenhack Thanks for explaining it. Should we install python 3.6 in the Travis container then?

My point is more like, instead of supporting `3.4` now, which it will be deprecated before 2020, we should look into 3.6 which we don't have to bother ourselves to do a version bump later since we are just about to set thing ups. Admittedly, I am not against using `3.4`

**Comment by zenhack** [comment]
The 3.x series isn't terribly variable, and 3.4 takes less set-up now.
By the time it's actually unsupported by the PSF, we can probably
upgrade to another version in travis just by bumping the number in our
config, whereas right now it's a (small) bit of work. If you want to do
that work, I'm happy to accept a patch.

Quoting Lucas H. Xu (2018-03-05 17:15:37)

**Comment by xuhang57** [comment]
I suggest we could merge this and close the issue #700 . Maybe open a new issue that is about supporting python 3. Or we could just keep using the old issue. 

Thanks Ian!

**Pull request merged**